design/kata-api-design.md

```diff
@@ -0,0 +1,143 @@
+# Kata API Design
+To fulfill the [kata design requirements](kata-design-requirements.md), and based on the disscusion on [Virtcontainers API extentions](https://docs.google.com/presentation/d/1dbGrD1h9cpuqAPooiEgtiwWDGCYhVPdatq7owsKHDEQ), the kata runtime library features following APIs:
+-  Sandbox based top API
+-  Storage and network hotplug API
+-  Plugin frameworks for external proprietary Kata runtime extensions
+-  Built-in shim and proxy types and capabilities
+
+## Sandbox Based API
+### Sandbox Management API
+-  CreateSandbox(ID, SandboxConfig)
+
+Create, start a sandbox and return the sandbox structure.
+-  ConnectSandbox(ID)
+
+Connect to an existing sandbox and return the sandbox structure.
+-  ListSandbox()
+
+List all existing standboxes with status.
+
+### Sandbox Operation API
+-  sandbox.Release()
+
+Release the sandbox connection associated with the sandbox structure.
+-  sandbox.Pause()
+
+Pause the sandbox.
+-  sandbox.Resume()
+
+Resume the paused sandbox.
+-  sandbox.Destroy()
+
+Destroy the sandbox and remove all persistent metadata.
+-  sandbox.Stats()
+
+Get the stats of the sandbox.
+-  sandbox.Done()
+
+Return a context handler that caller can monitor for sandbox callbacks such as
+error termination.
+-  sandbox.ListContainer()
```

Comment:
Ok cool. Then I think that we can remove it for now, and in the meantime, we could try to assess if `pod.Status()` is taking too much time because it's doing some extra things not needed when we want only the list of containers.